main.go

```diff
@@ -343,19 +344,15 @@ func mainerr() error {
 			case *fVersion:
 				fmt.Printf("%v %v\n", mp.Module.Path, mp.Module.Version)
 			case *fRun:
-				if runtime.GOOS == "windows" {
-					run := exec.Command(target, runArgs...)
-					run.Stdout = os.Stdout
-					run.Stderr = os.Stderr
-					if err := run.Run(); err != nil {
-						if _, ok := err.(*exec.ExitError); !ok {
-							return fmt.Errorf("failed to run %v: %v", strings.Join(run.Args, " "), err)
-						}
-					}
-				} else {
-					argv := append([]string{target}, runArgs...)
-					if err := syscall.Exec(argv[0], argv, os.Environ()); err != nil {
-						return fmt.Errorf("failed to exec %v: %v", strings.Join(argv, " "), err)
+				run := exec.Command(target, runArgs...)
+				run.Args[0] = path.Base(mp.ImportPath)
+				run.Stdout = os.Stdout
+				run.Stderr = os.Stderr
+				if err := run.Run(); err != nil {
+					if _, ok := err.(*exec.ExitError); ok {
+						return errRunExitErr
+					} else {
+						return fmt.Errorf("failed to run %v: %v", strings.Join(run.Args, " "), err)
```

Comment:
Given that the program failed to run at all, I wonder if it's worth printing all the args, which might be quite large, because the args are unlikely to have influenced the result.

Also, run.Args isn't that informative now that run.Args[0] has been replaced. How about:

    return fmt.Errorf("failed to run %v: %v", run.Path, err)

?